### ISSUE TITLE
sztp: parse the start of `conveyed-information` - rfc5652 (CMS)

### ISSUE THREAD

**Issue by glimchb** [opiproject/sztp#7]
```
$ jq -r .\"ietf-sztp-bootstrap-server:output\".\"conveyed-information\" /tmp/post_rpc_input.json | base64 --decode | tail -n +2 | sed  '1i {' | jq . | tee /tmp/post_rpc_fixed.json
{
  "ietf-sztp-conveyed-info:onboarding-information": {
    "boot-image": {
      "download-uri": [
        "http://web:8082/var/lib/misc/my-boot-image.img"
      ],
      "image-verification": [
        {
          "hash-algorithm": "ietf-sztp-conveyed-info:sha-256",
          "hash-value": "7b:ca:e6:ac:23:06:d8:79:06:8c:ac:03:80:e2:16:44:7e:40:6a:65:fa:d4:69:61:6e:05:ce:f5:87:dc:2b:97"
        }
      ]
    },
    "pre-configuration-script": "Iy9iaW4vYmFzaAplY2hvICJpbnNpZGUgdGhlIHByZS1jb25maWd1cmF0aW9uLXNjcmlwdC4uLiIK",
    "configuration-handling": "merge",
    "configuration": "PHRvcCB4bWxucz0iaHR0cHM6L2V4YW1wbGUuY29tL2NvbmZpZyI+CiAgPGFueS14bWwtY29udGVudC1va2F5Lz4KPC90b3A+Cg==",
    "post-configuration-script": "Iy9iaW4vYmFzaAplY2hvICJpbnNpZGUgdGhlIHBvc3QtY29uZmlndXJhdGlvbi1zY3JpcHQuLi4iCg=="
  }
}
```

BUT what happens on first 2 lines ?

```
$ jq -r .\"ietf-sztp-bootstrap-server:output\".\"conveyed-information\" /tmp/post_rpc_input.json | base64 --decode | head -n 2
0▒B
   *▒H▒▒+▒▒1▒-{
  "ietf-sztp-conveyed-info:onboarding-information": {
```

**Comment by glimchb** [comment]
in go:
```
diff --git a/sztp-agent/pkg/secureAgent/daemon.go b/sztp-agent/pkg/secureAgent/daemon.go
index 18949a7..e4705e1 100644
--- a/sztp-agent/pkg/secureAgent/daemon.go
+++ b/sztp-agent/pkg/secureAgent/daemon.go
@@ -12,6 +12,7 @@ import (
        "errors"
        "log"
        "os"
+       "github.com/github/smimesign/ietf-cms"
 )

 func (a *Agent) RunCommandDaemon() error {
@@ -60,5 +61,15 @@ func (a *Agent) doRequestBootstrapServer() error {
        }
        res.IetfSztpBootstrapServerOutput.ConveyedInformation = string(newVal)
        log.Println(res)
+       signedData, err := cms.ParseSignedData(newVal)
+       if err != nil {
+               return err
+       }
+       log.Println(signedData)
+       data, err := signedData.GetData()
+       if err != nil {
+               return err
+       }
+       log.Println(string(data))
        return nil
 }
(END)
```

**Comment by glimchb** [comment]
```
	"github.com/github/smimesign/ietf-cms"
       signedData, err := cms.ParseSignedData(newVal)
```
didn't work due to incorrect `ci.ContentType` , so replacing for now with
```
	"github.com/github/smimesign/ietf-cms/protocol"
	ci, err := protocol.ParseContentInfo(newVal)
```
see https://github.com/github/smimesign/blob/main/ietf-cms/oid/oid.go

**Comment by glimchb** [comment]
main fix in #59 
Additional fix (leftovers) in #62
